File: 성준/PG_타겟 넘버.py
from itertools import permutations
answer = 0

def solution(numbers, target):
    def dfs(cnt,ans):
        global answer
        if cnt >= len(numbers): #cnt가 numbers의 길이를 넘어가면 반복 종료
            if ans==target:
                answer+=1
            return 
        dfs(cnt+1,ans+numbers[cnt])
        dfs(cnt+1,ans-numbers[cnt])
        
    dfs(0,0)
    return answer

File: 홍준/PG_타겟넘버.java
class Solution {
    public int solution(int[] numbers, int target) {
        int answer = 0;
        int sum=0;
        int tmp=0;
        for(int i=0;i<numbers.length;i++){
            sum=sum+numbers[i];
        }
        for(int i=0;i<numbers.length;i++){
            tmp=sum;
            tmp=tmp-2*numbers[i];
            if(tmp==target){
                answer=answer+1;
                continue;
            }
            else if(tmp<target)
                continue;
            answer=find(numbers,target,i,tmp,answer);
        }
        return answer;
    }
    public static int find(int[] numbers,int target,int index,int tmp,int answer){
        int a=tmp;
        if(tmp==target)
            return answer+1;
        else if(tmp<target)
            return answer;
        for(int j=index+1;j<numbers.length;j++){
            a=tmp;
            a=a-2*numbers[j];
            answer=find(numbers,target,j,a,answer);
        }
        return answer;
    }
}
